### PR TITLE
added RankInfo backing field to return RankContainer Ranks or LegacyR…

### LINKED ISSUE #122: Add a "shortcut" to the RankContainer from SeasonalStats
I'll need to check exactly what season the rank types changed but it'd be cool if we had a property (with backing field) that could get the appropriate `RankContainer` for a user's rank.

Add it to the `SeasonalStats` class and name it `RankInfo` or something similar.

### PR DESCRIPTION
**Pull request by abuZayed15**
Closes #122 . @aspriddell Thank you for assigning this issue to me. I wasn't sure if this is what you were looking for. I wasn't sure about the JsonProperty data annotation either since I don't know if the json already has a property with `rank_info` that the program is fetching from. Please let me know and I would be happy to make any changes as quickly as possible.

Thank you for letting me work on this issue!

### CODE REVIEW

File: DragonFruit.Six.API/Data/SeasonStats.cs [hunk 1]
```diff
@@ -77,5 +78,16 @@ public class SeasonStats
 
         [JsonProperty("last_match_skill_stdev_change")]
         public double LastMatchSkillUncertaintyChange { get; set; }
+
+        [JsonProperty("rank_info")]
```
Comment (aspriddell) [inline DragonFruit.Six.API/Data/SeasonStats.cs:82]: don't need to add any json property as it's "computed"

File: DragonFruit.Six.API/Data/SeasonStats.cs [hunk 2]
```diff
@@ -77,5 +78,16 @@ public class SeasonStats
...
+        public RankContainer RankInfo
+        {
+            get
+            {
+                return Rank > 16 ? References.Ranks((int)Rank) : References.LegacyRanks((int)Rank);
+            }
+
+            set => value ??= Rank > 16 ? References.Ranks((int)Rank) : References.LegacyRanks((int)Rank);
```
Comment (aspriddell) [inline DragonFruit.Six.API/Data/SeasonStats.cs:90]: there's no need for get and set. It needs a backing field like:

```
private RankContainer _rankInfo;

public RankContainer RankInfo => _rankInfo ??= Rank > 16 ? References.Ranks((int)Rank) : References.LegacyRanks((int)Rank);
```

where the `private` backing field is at the top of the class (above all)

**Review by aspriddell — CHANGES_REQUESTED** [review]
there's also the cast to `int` that will need addressing, but that's for another PR

**Review by aspriddell — APPROVED** [review]
:+1:

File: DragonFruit.Six.API/Data/SeasonStats.cs (hunk 2 above)
Comment (abuZayed15) [reply DragonFruit.Six.API/Data/SeasonStats.cs]: Interesting. I am normally accustomed to only using a public property with getters and setters but this looks great. Thank you for showing this new type of assignment!

**Pull request merged**